### ISSUE TITLE
append functionality doesn't work

### ISSUE THREAD

**Issue by t1user** [ranaroussi/pystore#26]
I'm following the example notebook and after collection.append(...) the data is never added tot he store.

**Comment by cevans3098** [comment]
@t1user  do you have any code you can share - everything works fine for me

**Comment by t1user** [comment]
I'm literally using this notebook:
https://github.com/ranaroussi/pystore/blob/master/examples/pystore-tutorial.ipynb

when I get to:
```

collection.append('AAPL', aapl[-1:])

df = collection.item('AAPL').to_pandas()
df.tail()

```

the resulting dataframe doesn't have the last, appended row.

I'm on ubuntu 19.04, python 3.7.3 and my pip freeze is:

asn1crypto==0.24.0
attrs==19.1.0
backcall==0.1.0
bleach==3.1.0
certifi==2019.9.11
cffi==1.12.3
chardet==3.0.4
Click==7.0
cloudpickle==1.2.2
cryptography==2.7
dask==2.4.0
decorator==4.4.0
defusedxml==0.6.0
distributed==2.4.0
entrypoints==0.3
fastparquet==0.3.2
fsspec==0.4.4
HeapDict==1.0.1
idna==2.8
inflection==0.3.1
ipykernel==5.1.2
ipython==7.8.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
jedi==0.15.1
Jinja2==2.10.1
jsonschema==3.0.2
jupyter==1.0.0
jupyter-client==5.3.2
jupyter-console==6.0.0
jupyter-core==4.5.0
llvmlite==0.29.0
locket==0.2.0
MarkupSafe==1.1.1
mistune==0.8.4
more-itertools==5.0.0
msgpack==0.6.1
multitasking==0.0.9
nbconvert==5.6.0
nbformat==4.4.0
ndg-httpsclient==0.5.1
notebook==6.0.1
numba==0.45.1
numpy==1.17.2
pandas==0.25.1
pandocfilters==1.4.2
parso==0.5.1
partd==1.0.0
pexpect==4.7.0
pickleshare==0.7.5
pkg-resources==0.0.0
prometheus-client==0.7.1
prompt-toolkit==2.0.9
psutil==5.6.3
ptyprocess==0.6.0
pyasn1==0.4.7
pycparser==2.19
Pygments==2.4.2
pyOpenSSL==19.0.0
pyrsistent==0.15.4
PyStore==0.1.14
python-dateutil==2.8.0
python-snappy==0.5.4
pytz==2019.2
PyYAML==5.1.2
pyzmq==18.1.0
qtconsole==4.5.5
Quandl==3.4.8
requests==2.22.0
Send2Trash==1.5.0
six==1.12.0
sortedcontainers==2.1.0
tblib==1.4.0
terminado==0.8.2
testpath==0.4.2
thrift==0.11.0
toolz==0.10.0
tornado==6.0.3
traitlets==4.3.2
urllib3==1.25.3
wcwidth==0.1.7
webencodings==0.5.1
widgetsnbextension==3.5.1
zict==1.0.0

**Comment by cevans3098** [comment]
@t1user I believe there is a bug and it does not work properly with Dask 2.3 or 2.4.  see https://github.com/ranaroussi/pystore/issues/17#issuecomment-527209345
regarding this.  I have not upgraded to Dask 2.4 yet, but I know it does not work with Dask 2.3.  Try Dask 2.1 and it should work

@ranaroussi  any ideas on why Dask 2.3 does not work?  I have not spent a lot of time looking into the issue

**Comment by cheekybastard** [comment]
With the latest pystore: 0.1.14, using the pystore/examples/pystore-tutorial.ipynb,  I'm also finding that appends silently fails, no data appended. With dask: 2.6 to 2.2. 

Dask 2.1 when calling pystore append returns:

```
    /usr/local/anaconda/envs/dev/lib/python3.7/site-packages/dask/dataframe/core.py in     __getitem__(self, key)
   2671             return Series(graph, name, self._meta, self.divisions)
   2672         raise NotImplementedError(
-> 2673             "Series getitem in only supported for other series objects "
   2674             "with matching partition structure"
   2675         )

    NotImplementedError: Series getitem in only supported for other series objects with matching partition structure
```

**Comment by davidt35** [comment]
I'm getting same error when calling append function.  Series getitem in only supported for other series objects with matching partition structure

**Comment by davidt35** [comment]
Ok, it works now. First i upgraded numba to the latest version (i was still getting the error), then i installed dask 2.3.0 (i was still getting the error), but now it asked me to install ffspec, so i installed latest fspecc 0.5.2 and now it works.  Im running PyStore version PyStore==0.1.14.

my current setup:
```
PyStore==0.1.14
dask==2.3.0
numba==0.44.1
fsspec==0.5.2
Python 3.7.3
Windows 10
```

**Comment by cheekybastard** [comment]
Issue is the read_parquet in the append method, the exception not been surfaced is:
```

ValueError: The following columns were not found in the dataset {'index'}
The following columns were found Index(['Open', 'High', 'Low', 'Close', 'Volume', 'Ex-Dividend', 'Split Ratio',
       'Adj. Open', 'Adj. High', 'Adj. Low', 'Adj. Close', 'Adj. Volume'],
      dtype='object')
```

https://github.com/ranaroussi/pystore/blob/master/pystore/collection.py#L167

Change columns="index" to columns=[], will just read index.

```
old_index = dd.read_parquet(self._item_path(item, as_string=True),
                                        columns=[],
                                        engine=self.engine
                                        ).index.compute()
```

https://github.com/ranaroussi/pystore/blob/master/pystore/collection.py#L171

Add a raise to surface exceptions.
```
except Exception as error:
            raise ValueError("error: `%s`" % error)
```

**Comment by davidt35** [comment]
Works like a charm. Thank you.

**Comment by ranaroussi** [comment]
Fixed in 0.1.15. Please reopen issue if the problem persists.
